lib/core/client.js

```diff
@@ -863,7 +870,18 @@ function onSocketSession (session) {
 function connect (client) {
   assert(!client[kSocket])
 
-  const { protocol, port, hostname } = client[kUrl]
+  let { protocol, port, hostname } = client[kUrl]
+
+  // Resolve ipv6
+  if (hostname.startsWith('[')) {
+    const idx = hostname.indexOf(']')
+    const ip = idx !== -1 ? hostname.substr(1, idx - 1) /* istanbul ignore next */ : hostname
```

Comment:
remove ternary and just make an `assert(idx !== -1)`

Reply:
It should not because URL parsing would fail.
Ok, will do that!

Reply:
I'm fine with doing istanbul ignore or assertions.

Reply:
Done!